### PR TITLE
feature/CP-2045 added redirection to webview when there is no notific…

### PR DESCRIPTION
**Pull request by HardikLakum**
added redirection to WebView when there is no NotificationOpenedListener

### CODE REVIEW

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java [hunk 1]
```diff
@@ -1236,11 +1237,18 @@ public void removeNotificationReceivedListener() {
...
+        return fireNotificationOpenedListener(openedResult, null);
+    }
+
+    public boolean fireNotificationOpenedListener(final NotificationOpenedResult openedResult, Context context) {
         if (openedResult.getNotification().getAppBanner() != null) {
             showAppBanner(openedResult.getNotification().getAppBanner(), openedResult.getNotification().getId());
         }
 
         if (notificationOpenedListener == null) {
+            if(openedResult.getNotification().getUrl() != null && !openedResult.getNotification().getUrl().equalsIgnoreCase("")){
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/CleverPush.java:1249]: Code style (spaces & add new line)

File: cleverpush/src/main/java/com/cleverpush/NotificationOpenedProcessor.java [hunk 2]
```diff
@@ -32,18 +32,18 @@ public static void processIntent(Context context, Intent intent) {
...
-        cleverPush.fireNotificationOpenedListener(result);
-
         // open launcher activity
         Intent launchIntent = context.getPackageManager().getLaunchIntentForPackage(context.getPackageName());
         if (launchIntent != null) {
             launchIntent.setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT/* | Intent.FLAG_ACTIVITY_NEW_TASK*/);
             context.startActivity(launchIntent);
         }
 
+        CleverPush cleverPush = CleverPush.getInstance(context);
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/NotificationOpenedProcessor.java:42]: Is this change necessary? If not please revert.

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java (hunk 1 above)
Comment (gbhrdt) [reply cleverpush/src/main/java/com/cleverpush/CleverPush.java:1249]: Please also consider this:

https://github.com/cleverpush/cleverpush-android-sdk/blob/91f649d927a0c6c7cb2cdd4c36e01b6b4a0ccee8/cleverpush/src/main/java/com/cleverpush/CleverPush.java#L489

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java (hunk 1 above)
Comment (YasinY) [inline cleverpush/src/main/java/com/cleverpush/CleverPush.java:1249]: re use variable

Comment (YasinY) [reply cleverpush/src/main/java/com/cleverpush/CleverPush.java:1249]: openedResult.getNotification().getUrl()

**Review by YasinY — APPROVED** [review]

File: cleverpush/src/main/java/com/cleverpush/NotificationOpenedProcessor.java (hunk 2 above)
Comment (HardikLakum) [reply cleverpush/src/main/java/com/cleverpush/NotificationOpenedProcessor.java:42]: yes if I will open Webview before the launcher activity then it will be not visible to the user launcher activity will cover it

File: cleverpush/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -31,49 +33,43 @@
                 <action android:name="com.huawei.push.action.MESSAGING_EVENT" />
             </intent-filter>
         </service>
-
```
Comment (gbhrdt) [inline cleverpush/src/main/AndroidManifest.xml:34]: Pls merge master into your branch

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java [hunk 4]
```diff
@@ -2473,4 +2470,8 @@ public static void setIsSubscribeForTopicsDialog(boolean isSubscribeForTopicsDia
     public static void removeInstance() {
         instance = null;
     }
+
+    public void setWebViewOpenEnabled(boolean webViewOpenEnabled) {
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/CleverPush.java:2474]: Add Doc

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java [hunk 5]
```diff
@@ -221,7 +221,6 @@ public void init(@Nullable final NotificationOpenedListener notificationOpenedLi
 
     /**
      * initialize Cleverpush SDK with subscribed callback
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/CleverPush.java:223]: Why were these lines removed?

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java [hunk 6]
```diff
@@ -29,6 +29,7 @@
 import androidx.core.content.ContextCompat;
 
 import com.cleverpush.banner.AppBannerModule;
+import com.cleverpush.banner.WebViewActivity;
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/CleverPush.java:32]: Why does the WebViewActivity live under `banner ` package? In this case we use it for banners + for notifcations.
Please move it to somewhere else.

File: cleverpush/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -12,7 +12,9 @@
         <activity android:name=".stories.StoryDetailActivity"
             android:theme="@style/Theme.AppCompat.NoActionBar" />
 
-        <activity android:name=".banner.WebViewActivity" />
+        <activity android:name=".banner.WebViewActivity"
```
Comment (gbhrdt) [inline cleverpush/src/main/AndroidManifest.xml:15]: Why does the WebViewActivity live under `banner ` package? In this case we use it for banners + for notifcations.
Please move it to somewhere else.

File: cleverpush/src/main/java/com/cleverpush/NotificationOpenedProcessor.java (hunk 2 above)
Comment (gbhrdt) [reply cleverpush/src/main/java/com/cleverpush/NotificationOpenedProcessor.java:42]: Please fix indentation for the whole file

This change is still necessary and will not break users who do not use this feature? Please test.

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: cleverpush/src/main/java/com/cleverpush/NotificationOpenedProcessor.java (hunk 2 above)
Comment (HardikLakum) [reply cleverpush/src/main/java/com/cleverpush/NotificationOpenedProcessor.java:42]: yes this change is necessary as I have explained above and for the scenario in which users not using this feature will not break anything
